src/components/ebay-snackbar-dialog/test/mock/action-snackbar-template.marko

```diff
@@ -0,0 +1,23 @@
+<ebay-button on-click("openSnackbar")>
+    Open Action Snackbar
+</ebay-button>
+<ebay-snackbar-dialog open=state.open on-close('closeSnackbar') on-action("closeSnackbar")>
```

Comment:
We shouldn't need the developer to listen for close event and then set open state. We should handle this for them.

Reply:
Hmm I mean I did have it so the user didn't have to pass a close function, but then we talked about it more and I got rid of it because the user will basically always need to manage the state of the snackbar. 

In the action case, we will always need the developer to manage the state, because there are so many ways they might want to handle closing it. 

I agree though that with the no action basic example, we *could* have it so that the snackbar just needs to be passed the open boolean. But then if the user is already passing the boolean for open / close, it just makes sense to let them manage the state. 

The user would definitely need to at least pass the open / close boolean though, even in the basic state. Would they want it to start open and then close, or would they want it to trigger based on something and then auto close? It just seems much easier to have them manage the state. 

I know with material UI components when I was working with them, the user managed open / close states in a fair number of components. This is a case where that makes sense, and I know we have other components like that as well.

Reply:
By default, when user clicks the action button, toast should always close and set open to false. 
By default, after timeout, toast should always close and set open to false.
If the developer wants to do something else in *addition*, then yes they can tap into the on-close event to do that.

Can you compare with toast dialog? It doesn't seem to be requiring user to explicitly handle a close event.

Reply:
In the toast-dialog exmaple, it does require the user to have a close event. 

[example from toast dialog](https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-toast-dialog/examples/01-default/template.marko)

```marko
<ebay-toast-dialog a11y-close-text="Close Toast" open=state.open on-close("closeToast")>
    <@header>Heading</@header>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit</p>
    <p>
        <a href="http://www.ebay.com">www.ebay.com</a>
    </p>
    <@footer>
        <ebay-button accesskey="i" on-click("closeToast")>Close</ebay-button>
    </@footer>
</ebay-toast-dialog>
<ebay-button on-click("openToast")>Open Toast</ebay-button>
class {
    onCreate() {
        this.state = { open: false };
    }
    openToast() {
        this.state.open = true;
    }
    closeToast() {
        this.state.open = false;
    }
}
```

Reply:
Hmm, yes. It's done differently though. It is closed via a click event on the ebay-button, which makes more sense to me than a close event on the snackbar itself. Calling a close function on the close event of the snackbar is what struck me as odd initially. Can we add event listeners to attribute tags?

Reply:
Hmm, this doesn't seem right: `on-close("closeToast")>`. Let's check in tomorrow with Andrew.

Reply:
So after some offline discussion it might be better to rename `on-close("closeToast")` to `on-close("handleClose")` to make it clearer that this is something that an application _can_ do to synchronize its state with the components current state, but is not necessary to actually close the toast. Likewise for snackbar.